backend-es/test/snapshots-out/Snapshot.TransformerReaderT01.js

```diff
@@ -0,0 +1,118 @@
+  const a$p$2 = Effect$dRandom.randomInt(1)(10)();
+  const a$1 = Effect$dRandom.randomInt(10)(20)();
+  return ((((4 + a | 0) + (a$p + 1 | 0) | 0) + (a$p$1 + a$p$2 | 0) | 0) + 5 | 0) + a$1 | 0;
+};
+const program2 = dictMonadEffect => {
+  const Monad0 = dictMonadEffect.Monad0();
+  const Bind1 = Monad0.Bind1();
+  const bindReaderT1 = Control$dMonad$dReader$dTrans.bindReaderT(Bind1);
+  const liftEffect1 = Control$dMonad$dReader$dTrans.monadEffectReader(dictMonadEffect).liftEffect;
+  const Apply0 = Bind1.Apply0();
+  const $6 = Apply0.Functor0();
+  const $7 = Monad0.Applicative0().pure;
+  const local1 = Control$dMonad$dReader$dTrans.monadReaderReaderT(Monad0).local;
+  const $9 = Monad0.Applicative0();
+  return bindReaderT1.bind(liftEffect1(Effect$dConsole.log("foo")))(() => bindReaderT1.bind(liftEffect1(Effect$dRandom.randomInt(1)(10)))(i1 => bindReaderT1.bind((() => {
+    const $12 = $6.map(v => v + 1 | 0);
+    const $13 = liftEffect1(Effect$dRandom.randomInt(1)(10));
+    return x => $12($13(x));
+  })())(i2 => bindReaderT1.bind((() => {
+    const $13 = $6.map(Data$dSemiring.intAdd);
+    const $14 = liftEffect1(Effect$dRandom.randomInt(1)(10));
+    const $15 = liftEffect1(Effect$dRandom.randomInt(1)(10));
+    return r => Apply0.apply($13($14(r)))($15(r));
+  })())(i3 => bindReaderT1.bind($7)(five => bindReaderT1.bind(local1(v => v * 2 | 0)(bindReaderT1.bind($7)(ten => liftEffect1(Effect$dRandom.randomInt(ten)(20)))))(i4 => {
+    const $16 = $9.pure(((((4 + i1 | 0) + i2 | 0) + i3 | 0) + five | 0) + i4 | 0);
+    return v => $16;
+  }))))));
+};
+const test3 = /* #__PURE__ */ (() => {
+  const bindReaderT1 = Control$dMonad$dReader$dTrans.bindReaderT(Effect.bindEffect);
+  const liftEffect1 = Control$dMonad$dReader$dTrans.monadEffectReader(Effect$dClass.monadEffectEffect).liftEffect;
+  const local1 = Control$dMonad$dReader$dTrans.monadReaderReaderT(Effect.monadEffect).local;
+  return bindReaderT1.bind(liftEffect1(Effect$dConsole.log("foo")))(() => bindReaderT1.bind(liftEffect1(Effect$dRandom.randomInt(1)(10)))(i1 => bindReaderT1.bind((() => {
+    const $5 = liftEffect1(Effect$dRandom.randomInt(1)(10));
+    return x => {
+      const $7 = $5(x);
+      return () => {
+        const a$p = $7();
+        return a$p + 1 | 0;
+      };
+    };
+  })())(i2 => bindReaderT1.bind((() => {
+    const $6 = liftEffect1(Effect$dRandom.randomInt(1)(10));
+    const $7 = liftEffect1(Effect$dRandom.randomInt(1)(10));
+    return r => {
+      const $9 = $6(r);
+      const $10 = $7(r);
+      return () => {
+        const a$p = $9();
+        const a$p$1 = $10();
+        return a$p + a$p$1 | 0;
+      };
+    };
+  })())(i3 => bindReaderT1.bind(Effect.pureE)(five => bindReaderT1.bind(local1(v => v * 2 | 0)(bindReaderT1.bind(Effect.pureE)(ten => liftEffect1(Effect$dRandom.randomInt(ten)(20)))))(i4 => {
+    const $9 = ((((4 + i1 | 0) + i2 | 0) + i3 | 0) + five | 0) + i4 | 0;
+    return v => () => $9;
+  }))))))(5);
+})();
```

Comment:
Is it possible for this to become as good as `test1`? And likewise for `test3`?

I renamed the other functions to `programX` to better illustrate what's actually being tested.

Reply:
I think that we should add a rule that an external app followed by an accessor should always be inlined.

Reply:
Actually, I don't think that would work in some of these cases. I guess we need to apply the same directive inference rules for top-level applications to local bindings :/.